cartoframes/data/observatory/catalog/repository/repo_client.py

```diff
@@ -1,100 +1,83 @@
-
-        return self._run_query(query, filters, extra_condition)
-
-    def _run_query(self, query, filters, extra_conditions=None):
-        conditions = self._compute_conditions(filters, extra_conditions)
-
-        if len(conditions) > 0:
-            where_clause = ' AND '.join(conditions)
-            query += ' WHERE {}'.format(where_clause)
-
-        return self._context_manager.execute_query(query).get('rows')
+        return self._get_entity('datasets', filters, use_slug=True)
 
-    def _compute_conditions(self, filters, extra_conditions):
-        conditions = extra_conditions or []
-
-        if filters is not None and len(filters) > 0:
-            conditions.extend([self._generate_condition(key, value) for key, value in sorted(filters.items())])
-
-        return conditions
-
-    @staticmethod
-    def _generate_condition(key, value):
-        if isinstance(value, list):
-            value_list = ','.join(["'" + v + "'" for v in value])
-            return "t.{} IN ({})".format(key, value_list)
+    def get_geographies(self, filters=None):
+        return self._get_entity('geographies', filters, use_slug=True)
 
-        return "t.{} = '{}'".format(key, value)
+    def get_variables(self, filters=None):
+        filter_id = self._get_filter_id(filters, use_slug=True)
+        if filter_id:
+            return self._fetch_entity_id('variables', filter_id)
+        else:
+            entity = 'datasets/{}/variables'.format(filters.pop('dataset'))
+            return self._fetch_entity(entity, filters)
 
-    def __new__(cls):
-        if not RepoClient.__instance:
-            RepoClient.__instance = object.__new__(cls)
-        return RepoClient.__instance
+    def get_variables_groups(self, filters=None):
+        filter_id = self._get_filter_id(filters, use_slug=True)
+        if filter_id:
+            return self._fetch_entity_id('variables_groups', filter_id)
+        else:
+            entity = 'datasets/{0}/variables_groups'.format(filters.pop('dataset'))
+            return self._fetch_entity(entity, filters)
+
+    def _get_filter_id(self, filters, use_slug=False):
+        if isinstance(filters, dict):
+            filter_id = filters.get('id')
+            if not filter_id and use_slug:
+                filter_id = filters.get('slug')
+            return filter_id
+
+    def _get_entity(self, entity, filters=None, use_slug=False):
+        filter_id = self._get_filter_id(filters, use_slug)
+        if filter_id:
+            return self._fetch_entity_id(entity, filter_id)
+        else:
+            return self._fetch_entity(entity, filters)
+
+    def _fetch_entity_id(self, entity, filter_id):
+        if isinstance(filter_id, list):
+            return list(filter(None, [self._fetch_entity(os.path.join(entity, _id)) for _id in filter_id]))
+        else:
+            return self._fetch_entity(os.path.join(entity, filter_id))
+
+    def _fetch_entity(self, entity, filters=None):
+        if self._do_dataset:
```

Comment:
Just a question.

Now the repo_client uses the default DO user (do-metadata) to retrieve most of the information.

Only when getting the subscriptions (Datasets and Geographies) the user credentials are used. But, since we already have the list of subscribed ids, what is the reason of using the user credentials instead of the default public user for the request?

Reply:
> Only when getting the subscriptions (Datasets and Geographies) the user credentials are used.

We should use them if credentials exists (for metric purposes for example)

Reply:
OK, just for the metrics. :+1: